core/src/main/scala/org/genivi/sota/core/daemon/DeltaListener.scala

```diff
@@ -0,0 +1,61 @@
+package org.genivi.sota.core.daemon
+
+import akka.Done
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.core.UpdateService
+import org.genivi.sota.core.campaigns.CampaignLauncher
+import org.genivi.sota.core.db.Campaigns
+import org.genivi.sota.data.PackageId
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages.{GeneratedDelta, GeneratingDeltaFailed}
+import slick.driver.MySQLDriver.api._
+import eu.timepit.refined._
+import org.genivi.sota.marshalling.{DeserializationException, RefinementError}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeltaListener(deviceRegistry: DeviceRegistry, updateService: UpdateService,
+                    messageBus: MessageBusPublisher)(implicit db: Database)  {
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  def generatedDeltaAction(msg: GeneratedDelta)(implicit ec: ExecutionContext): Future[Done] = {
+    val fromR = refineV[PackageId.ValidVersion](msg.from) match {
+                  case Left(e) => throw DeserializationException(RefinementError(msg.from, e))
+                  case Right(v) => v
+                }
```

Comment:
> Yikes, throwing exceptions, that is so 90-ties we are in the `Future` now.

On a more serious note, why not change the type of the Event to have the from be `PackageId.Version` from the beginning?